src/read.c

```diff
@@ -4257,21 +4267,31 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
             maxItemID = item->id;
         }
         if (item->dimgForID == colorItem->id) {
+            avifBool seenAlphaForCurrentItem = AVIF_FALSE;
             for (uint32_t j = 0; j < meta->items.count; ++j) {
                 avifDecoderItem * auxlItem = &meta->items.item[j];
                 if (avifDecoderItemIsAlphaAux(auxlItem, item->id)) {
+                    if (seenAlphaForCurrentItem) {
+                        // Multiple items are claiming to be the alpha auxiliary of the current item.
+                        avifFree(alphaItemIndices);
+                        *alphaItem = NULL;
+                        *isAlphaItemInInput = AVIF_FALSE;
+                        return AVIF_RESULT_INVALID_IMAGE_GRID;
+                    }
                     alphaItemIndices[alphaItemCount++] = j;
+                    seenAlphaForCurrentItem = AVIF_TRUE;
                 }
             }
+            if (!seenAlphaForCurrentItem) {
+                // No alpha auxiliary item was found for the current item. Treat this as an image without alpha.
```

Comment:
Could you remind me why we should treat this as an image without alpha, rather than return an error?

Reply:
When the original change was made, that was libavif's behavior. And there seems to be some test files this way. So let's not change the behavior in this PR.

I am okay treating this as an error, but we should update the test files to ensure we don't break anything.